src/Sprites/ImageHeader.cpp

```diff
@@ -54,3 +54,29 @@ void ImageHeader::VerifyIndexedBitCount(uint16_t bitCount)
 		throw std::runtime_error("A bit count of " + std::to_string(bitCount) + " does not support an indexed palette");
 	}
 }
+
+void ImageHeader::Validate()
+{
+	if (headerSize != sizeof(ImageHeader)) {
+		throw std::runtime_error("Image Header must be equal to " + std::to_string(sizeof(ImageHeader)));
+	}
+
+	if (planes != defaultPlanes) {
+		throw std::runtime_error("Planes must be equal to " + std::to_string(defaultPlanes));
+	}
+
+	VerifyValidBitCount(bitCount);
+
+	auto compressionCast = static_cast<uint32_t>(compression);
+	if (compressionCast > 13) {
+		throw std::runtime_error("A compression value of " + std::to_string(compressionCast) + " is invalid");
```

Comment:
As the format could be expanded, and new compression codes added, I think this should be about lack of support.

Reply:
I would like to leave information on forcing the compression to be 0 (uncompressed) out of ImageHeader as this is our implementation concern and not specific to an ImageHeader. Would you prefer this check just be removed? I'm okay either way. 

It was also sort of an experiment for me to see what would happen if I casted an integer outside of the scoped enum's range. C++ doesn't mind. I think in C# this would have caused an exception, but I would need to check for sure.

-Brett

Reply:
Hmm, if we just copy the pixel data around, it's probably fine if we don't understand the compression code. If we access the pixel data in any way at all, then it should be a compression format that we support.

I think we should avoid making assumptions about what codes are valid as per the specification, as the specification can be updated and extended. Writing the message as "is invalid" seems to suggest it is outside the specification of the file format, but we don't know for certain the specification won't change.

----

I suppose the question here is what do the validity checks represent. Are they checks that must be true for all bitmap files to be valid and conform to the specification, or are they about what we support. Sounds like you want this to be about the specification itself, rather than what we support. In which case, we should be more careful about what specification we are validating against. That is, we should include some kind of version number for the spec, and name the validate method accordingly. If we want to get really picky, even the size of the header structure can change between versions. That is one of the ways it determines which version the file format is. If you want to make this solely about the specification, than we may need to adjust the header size check, the plane check, and the compression code check. 

I would caution against checking for validity against the spec, since that could easily turn into a big hairy beast. I'm not prepared to put that much effort into understanding the intricacies of the evolution of the bitmap file format as it's progressed over the years from OS/2 to modern versions of Windows. I'm not even sure it's fully documented anywhere.

----

We should have some checks somewhere about what we support. If we only support uncompressed, and throw an error if we encounter any other type, I'm fine with that. That would be easy enough to do.

If we want to work a little harder at it, the error might be delayed to accessing the pixel data in a manner other than copying it all around as one big chunk. I'm not sure we need or want to go there.

----

I think I'm fine with keeping this check somewhere, but we may be going down a path that is much more work than we want to do. Your call.

If you keep the check, I suggest rewording the "is invalid" to "is unsupported".

Reply:
I'll delete the check here and it can be handled int he Reader/Writer as it is encountered based on what is supported.

-Brett